src/PluginClientFactoryInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Http\Client\Common;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+
+/**
+ * Create a PluginClient.
+ *
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ */
+interface PluginClientFactoryInterface
+{
+    /**
+     * @param HttpClient|HttpAsyncClient $client
+     * @param Plugin[]                   $plugins
+     * @param array                      $options
+     *
+     * @see PluginClient constructor for $options details.
```

Comment:
I don't think so. There is only one PluginClient without a dedicated interface and this implementation is final. There is no way to write an other PluginClient!